From 7293c0528e3c2328eae9339d5bb1fa19e91c21eb Mon Sep 17 00:00:00 2001
From: Michael Tretter <m.tretter@pengutronix.de>
Date: Fri, 17 Sep 2021 09:51:51 +0200
Subject: [PATCH] Add a boot.json file to load binaries in Litex

The boot.json file allows to load the kernel, dtb, and initramfs as
separate binaries, which makes compilation and updates easier.
---
 boot.json       | 7 +++++++
 conf/ecpix5.dts | 6 +++++-
 2 files changed, 12 insertions(+), 1 deletion(-)
 create mode 100644 boot.json

diff --git a/boot.json b/boot.json
new file mode 100644
index 000000000000..f39452467b45
--- /dev/null
+++ b/boot.json
@@ -0,0 +1,7 @@
+{
+    "Image": "0x80200000",
+    "ecpix5.dtb": "0x81000000",
+    "r2": "0x81000000",
+    "rootfs.cpio": "0x82000000",
+    "bbl.bin": "0x80000000"
+}
diff --git a/conf/ecpix5.dts b/conf/ecpix5.dts
index 88374d682516..84c4cac6aad6 100644
--- a/conf/ecpix5.dts
+++ b/conf/ecpix5.dts
@@ -6,7 +6,11 @@
 	compatible = "freechips,rocketchip-unknown-dev";
 	model = "freechips,rocketchip-unknown";
 	chosen {
-		bootargs = "earlycon=sbi console=liteuart swiotlb=noforce";
+		bootargs = "earlycon=sbi console=liteuart swiotlb=noforce root=/dev/ram0";
+		riscv,kernel-start = <0x80200000>;
+		riscv,kernel-end = <0x81000000>;
+		linux,initrd-start = <0x82000000>;
+		linux,initrd-end = <0x84000000>;
 	};
 	L13: cpus {
 		#address-cells = <1>;
